### PR TITLE
feat: handle python input synchronously

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

When putting this together I was lucky enough to find the discussion on https://github.com/pyodide/pyodide/issues/1219. Part of that discussion was about how to use synchronous http requests to block (credit to: https://jasonformat.com/javascript-sleep/) and therefore mimic the behaviour of Python's `input` in JS.

Prior to this PR, we had to make `input` async, which automatically meant any function that called it had to be async. This proved challenging, since it was hard to correctly transform all the code to use async await, let alone do so without altering the code's behaviour.

I still have a fair bit of cleanup to do, hence this being in draft, but I think this is good enough to test out.

### CODE REVIEW

File: client/src/templates/Challenges/utils/build.ts [hunk 1]
```diff
@@ -163,13 +168,38 @@ export function getTestRunner(
...
 ) {
   const code = {
     contents: sources.index,
-    editableContents: sources.editableContents
+    editableContents: sources.editableContents,
+    original: sources.original
   };
 
-  const testWorker = createWorker(testEvaluator, { terminateWorker: true });
+  // TODO: allow terminateWorker to be passed in as an option.
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/utils/build.ts:201]:
```suggestion
```
Already done.

File: tools/client-plugins/browser-scripts/python-test-evaluator.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+  ctx.postMessage({ type: 'contentLoaded' });
+
+  return pyodide;
+}
+
+void setupPyodide();
+
+ctx.onmessage = async (e: PythonRunEvent) => {
+  console.log('python worker received message', e.data);
```
Comment (ShaunSHamilton) [inline tools/client-plugins/browser-scripts/python-test-evaluator.ts:57]:
```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Nice job getting this working!

I got a few type errors in the `build.ts` file, and could not figure out if this was just my LSP doing what it does or an actual error. Could someone double-check whether they are still there?

**Comment by ojeytonwilliams** [comment]
> Nice job getting this working!

Thanks, it was a bit of a struggle!

> I got a few type errors in the `build.ts` file, and could not figure out if this was just my LSP doing what it does or an actual error. Could someone double-check whether they are still there?

If you're seeing errors in `composeFunctions` and when `checkFilesErrors` is called, those are pre-existing errors.

**Review by ShaunSHamilton — APPROVED** [review]
I am not breaking this with anything. So, I think we are GTMAP 👍

**Review by naomi-lgbt — APPROVED** [review]
image

I did a python and it worked.

**Comment by ojeytonwilliams** [comment]
Thanks Naomi!

**Pull request merged**